src/geode/geosciences/pl_output.cpp

```diff
@@ -0,0 +1,182 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/geosciences/private/pl_output.h>
+
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <geode/basic/attribute_manager.h>
+
+#include <geode/basic/filename.h>
+#include <geode/mesh/core/edged_curve.h>
+
+#include <geode/geosciences/private/gocad_common.h>
+
+namespace
+{
+    class PLOutputImpl
+    {
+    public:
+        static constexpr geode::index_t OFFSET_START{ 1 };
+        static constexpr char EOL{ '\n' };
+        static constexpr char SPACE{ ' ' };
+        PLOutputImpl(
+            absl::string_view filename, const geode::EdgedCurve3D& edged_curve )
+            : file_( filename.data() ), edged_curve_( edged_curve )
+        {
+            OPENGEODE_EXCEPTION(
+                file_.good(), "Error while opening file: ", filename );
+        }
+
+        void write_prop_header()
+        {
+            const auto names =
+                edged_curve_.vertex_attribute_manager().attribute_names();
+            geode::detail::PropHeaderData prop_header;
+            std::vector< geode::detail::PropClassHeaderData > prop_class_header;
+
+            for( const auto& name : names )
+            {
+                const auto attribute = edged_curve_.vertex_attribute_manager()
+                                           .find_generic_attribute( name );
+                if( !attribute || !attribute->is_genericable() )
+                {
+                    continue;
+                }
+                generic_att_.push_back( attribute );
+                prop_header.names.push_back( name.data() );
+                prop_header.prop_legal_ranges.push_back(
+                    std::make_pair( "**none**", "**none**" ) );
+                prop_header.no_data_values.push_back( -99999. );
+                prop_header.property_classes.push_back( name.data() );
+                prop_header.kinds.push_back( " Real Number" );
+                prop_header.property_subclass.push_back(
+                    std::make_pair( "QUANTITY", "Float" ) );
+                prop_header.esizes.push_back( 1 );
+                prop_header.units.push_back( "unitless" );
+
+                geode::detail::PropClassHeaderData tmp_prop_class_header;
+                tmp_prop_class_header.name = name.data();
+                tmp_prop_class_header.kind = " Real Number";
+                tmp_prop_class_header.unit = "unitless";
+                tmp_prop_class_header.is_z = false;
```

Comment:
It looks the default attribute memeber values. Maybe simply add a constructor with the name (which is mandotory) and then do `prop_class_header.emplace_back( name );`